packages/cloud/src/utils.ts

```diff
@@ -38,3 +57,90 @@ export function parseFromStringArrayAsync(
     resolve(results);
   });
 }
+
+export const logCombatStatsAsync = async (combat: AtomicArenaCombat) => {
+  if (!SQLDB.isInitialized) {
+    await SQLDB.initialize();
+  }
+
+  const averageMMR =
```

Comment:
Maybe it makes sense to pass the stub as well into this function to avoid recalculating some of these

Reply:
We could consider writing all of these to the stub directly and then just passing the stub; then we'd have the data in Firestore as well if we ever needed it

Reply:
oh interesting. let me fiddle around that a bit.

Reply:
actually it looks like most of the heavy calculations are done at team/player level, and the stub doesn't have that structure.. it would be a significant addition to have all of them on the stub. For now I think we'll just pass stub in and reuse what's already there, and consider moving more logic to stub when we need them.